src/script/interpreter.cpp

```diff
@@ -1452,11 +1535,25 @@ void PrecomputedTransactionData::Init(const T& txTo, std::vector<CTxOut>&& spent
         if (uses_bip341_taproot && uses_bip143_segwit) break; // No need to scan further if we already need all.
     }
 
-    if (uses_bip143_segwit || uses_bip341_taproot) {
+    if (uses_bip143_segwit || uses_bip341_taproot || uses_bip119_ctv) {
         // Computations shared between both sighash schemes.
         m_prevouts_single_hash = GetPrevoutsSHA256(txTo);
         m_sequences_single_hash = GetSequencesSHA256(txTo);
         m_outputs_single_hash = GetOutputsSHA256(txTo);
+
+        bool skip_scriptSigs = std::find_if(txTo.vin.begin(), txTo.vin.end(),
+                [](const CTxIn& c) { return c.scriptSig != CScript(); }) == txTo.vin.end();
+        if (skip_scriptSigs) {
+            // 0 hash used to signal if we should skip scriptSigs
+            // when re-computing for different indexes.
+            m_scriptSigs_single_hash = uint256{};
+            // TODO: Cache midstate?
+            m_standard_template_single_hash = GetStandardTemplateHashEmptyScript(txTo, m_outputs_single_hash, m_sequences_single_hash, 0);
+        } else {
+            m_scriptSigs_single_hash = GetScriptSigsSHA256(txTo);
+            m_standard_template_single_hash = GetStandardTemplateHashWithScript(txTo, m_outputs_single_hash, m_sequences_single_hash, m_scriptSigs_single_hash, 0);
```

Comment:
Would it be safer to set `input_index` here to `uint32_t`'s max value? That way if there is less of a chance that we miss in a test the handling of a changing `input_index` since most likely the tested `input_index` will be `0`

Reply:
We do lose the effiency of always having the `0`th input cached however

Reply:
yeah the whole point is caching the most common case -- we could also cache the midstate without index and just finalize with an index in consensus.